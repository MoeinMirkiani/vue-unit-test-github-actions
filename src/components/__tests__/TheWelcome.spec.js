import { describe, it, expect } from "vitest"
import Welcome from "../TheWelcome.vue"
import { mount } from "@vue/test-utils"
describe("Welcome", () => {
    it('should render properly', () => {
        const wrapper = mount(Welcome)
        expect(wrapper.html()).toMatchSnapshot()
    })
})
