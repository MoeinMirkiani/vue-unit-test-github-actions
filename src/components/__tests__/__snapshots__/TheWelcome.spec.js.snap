// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Welcome > should render properly 1`] = `
"<div data-v-bd9b3c35="" class="item" welcometext="Hello">
  <div data-v-bd9b3c35="" class="details">
    <h3 data-v-bd9b3c35="">Documentation</h3> Vueâ€™s <a href="https://vuejs.org/" target="_blank" rel="noopener">official documentation</a> provides you with all information you need to get started.
  </div>
</div>
<div data-v-bd9b3c35="" class="item">
  <div data-v-bd9b3c35="" class="details">
    <h3 data-v-bd9b3c35="">Tooling</h3> This project is served and bundled with <a href="https://vite.dev/guide/features.html" target="_blank" rel="noopener">Vite</a>. The recommended IDE setup is <a href="https://code.visualstudio.com/" target="_blank" rel="noopener">VSCode</a> + <a href="https://github.com/johnsoncodehk/volar" target="_blank" rel="noopener">Volar</a>. If you need to test your components and web pages, check out <a href="https://www.cypress.io/" target="_blank" rel="noopener">Cypress</a> and <a href="https://on.cypress.io/component" target="_blank" rel="noopener">Cypress Component Testing</a>. <br> More instructions are available in <code>README.md</code>.
  </div>
</div>
<div data-v-bd9b3c35="" class="item">
  <div data-v-bd9b3c35="" class="details">
    <h3 data-v-bd9b3c35="">Ecosystem</h3> Get official tools and libraries for your project: <a href="https://pinia.vuejs.org/" target="_blank" rel="noopener">Pinia</a>, <a href="https://router.vuejs.org/" target="_blank" rel="noopener">Vue Router</a>, <a href="https://test-utils.vuejs.org/" target="_blank" rel="noopener">Vue Test Utils</a>, and <a href="https://github.com/vuejs/devtools" target="_blank" rel="noopener">Vue Dev Tools</a>. If you need more resources, we suggest paying <a href="https://github.com/vuejs/awesome-vue" target="_blank" rel="noopener">Awesome Vue</a> a visit.
  </div>
</div>
<div data-v-bd9b3c35="" class="item">
  <div data-v-bd9b3c35="" class="details">
    <h3 data-v-bd9b3c35="">Community</h3> Got stuck? Ask your question on <a href="https://chat.vuejs.org" target="_blank" rel="noopener">Vue Land</a>, our official Discord server, or <a href="https://stackoverflow.com/questions/tagged/vue.js" target="_blank" rel="noopener">StackOverflow</a>. You should also subscribe to <a href="https://news.vuejs.org" target="_blank" rel="noopener">our mailing list</a> and follow the official <a href="https://twitter.com/vuejs" target="_blank" rel="noopener">@vuejs</a> twitter account for latest news in the Vue world.
  </div>
</div>
<div data-v-bd9b3c35="" class="item">
  <div data-v-bd9b3c35="" class="details">
    <h3 data-v-bd9b3c35="">Support Vue</h3> As an independent project, Vue relies on community backing for its sustainability. You can help us by <a href="https://vuejs.org/sponsor/" target="_blank" rel="noopener">becoming a sponsor</a>.
  </div>
</div>"
`;
